src/shogun/regression/LinearRidgeRegression.cpp

```diff
@@ -42,23 +44,18 @@ void CLinearRidgeRegression::init()
 
 bool CLinearRidgeRegression::train_machine(CFeatures* data)
 {
-	if (!m_labels)
-		SG_ERROR("No labels set\n")
+	REQUIRE(m_labels,"No labels set\n");
 
 	if (!data)
 		data=features;
 
-	if (!data)
-		SG_ERROR("No features set\n")
+	REQUIRE(data,"No features set\n");
 
-	if (m_labels->get_num_labels() != data->get_num_vectors())
-		SG_ERROR("Number of training vectors does not match number of labels\n")
+	REQUIRE(m_labels->get_num_labels() == data->get_num_vectors(),"Number of training vectors does not match number of labels\n");
 
-	if (data->get_feature_class() != C_DENSE)
-		SG_ERROR("Expected Dense Features\n")
+	REQUIRE(data->get_feature_class() == C_DENSE,"Expected Dense Features\n");
```

Comment:
for your comments about the error messages , I just copied them from similar checks, I wanted the messages to be consistent with the others.